### ISSUE TITLE
Upgrading 7.0.1 to 7.0.4 fail (Invalid cross-device link)

### ISSUE THREAD

**Issue by LoicMahieu** [owncloud/apps#1997]
From docker ( https://registry.hub.docker.com/u/jchaney/owncloud/ ), I run a owncloud server instance in 7.0.1. I want to upgrade to 7.0.4 via the updater and I get this error:
```
Error   updater     Unable to move /var/www/owncloud/3rdparty/Archive to /var/www/owncloud/_oc-upgrade/tmp/3rdparty/Archive
Error   updater     Something went wrong for 3rdparty. Rolling back.
Error   PHP         rename(/var/www/owncloud/3rdparty/Archive,/var/www/owncloud/_oc-upgrade/tmp/3rdparty/Archive): Invalid cross-device link at /var/www/owncloud/apps/updater/lib/helper.php#27
Error   PHP         rename(/var/www/owncloud/3rdparty/Archive,/var/www/owncloud/_oc-upgrade/tmp/3rdparty/Archive): Invalid cross-device link at /var/www/owncloud/apps/updater/lib/helper.php#27
Error   PHP         rename(): The first argument to copy() function cannot be a directory at /var/www/owncloud/apps/updater/lib/helper.php#27
```

Only `/var/www/owncloud/data` is a docker volume. Original docker run:
```
docker run \
  -d \
  -p 443:443 \
  --name owncloud-server \
  -v /home/owncloud/data:/var/www/owncloud/data \
  -v /home/owncloud/ssl:/root/ssl \
  -e "SSL_KEY=/root/ssl/owncloud.key" \
  -e "SSL_CERT=/root/ssl/owncloud.crt" \
  jchaney/owncloud
```

**Comment by RealRancor** [comment]
Hi,

probably should be also re-opened at:

https://github.com/owncloud/updater/issues

**Comment by LoicMahieu** [comment]
Docker automated build would do the update job! No need to get the updater works in fact... Just replace the container.

**Comment by VicDeo** [comment]
it's just one of options like using a distro package manager. 
Cross partition move for directories is actually a PHP bug.
Fixed in Updater shipped with OC 7.0.3
